### ISSUE TITLE
esplugin_mysql.sql is missing

### ISSUE THREAD

**Issue by JasonLee120606** [kanersps/esplugin_mysql#9]

**Comment by JasonLee120606** [comment]
my esplugin_mysql.sql is missing any one can help?

**Comment by kanersps** [comment]
For this new version you only need the resource: fx migrant. No sql files are needed

**Comment by phongsmod1** [comment]
> For this new version you only need the resource: fx migrant. No sql files are needed
> […](#)
> Op 26 mei 2019 om 14:30 heeft JasonLee120606 ***@***.***> het volgende geschreven: my esplugin_mysql.sql is missing any one can help? — You are receiving this because you are subscribed to this thread. Reply to this email directly, view it on GitHub, or mute the thread.

pls tell me Where is resource: fx migrant ??

**Comment by Rikydc01** [comment]
So, what to do if before that we had to import the .sql file in my database? Now what i have to do?
![errore](https://user-images.githubusercontent.com/49121547/58405655-3a9f8e00-8068-11e9-9107-af999f20f233.png)

**Comment by kanersps** [comment]
Just start:

https://github.com/blattersturm/fxmigrant

Before starting esplugin_mysql and it will do everything for you!

**Comment by Rikydc01** [comment]
And what about the database? Which file do i have to import?

**Comment by Big-Daddy-Grimm** [comment]
I am having the same problem, get the error and can't proceed with setting up

**Comment by kanersps** [comment]
Nothing changed. You only need to create a database named essentialmode and run the resource. It will handle the database automatically

**Comment by Big-Daddy-Grimm** [comment]
run this

**Comment by Big-Daddy-Grimm** [comment]
-- --------------------------------------------------------
-- Host:                         127.0.0.1
-- Server version:               10.1.22-MariaDB - mariadb.org binary distribution
-- Server OS:                    Win64
-- HeidiSQL Version:             9.4.0.5125
-- --------------------------------------------------------

/*!40101 SET @OLD_CHARACTER_SET_CLIENT=@@CHARACTER_SET_CLIENT */;
/*!40101 SET NAMES utf8 */;
/*!50503 SET NAMES utf8mb4 */;
/*!40014 SET @OLD_FOREIGN_KEY_CHECKS=@@FOREIGN_KEY_CHECKS, FOREIGN_KEY_CHECKS=0 */;
/*!40101 SET @OLD_SQL_MODE=@@SQL_MODE, SQL_MODE='NO_AUTO_VALUE_ON_ZERO' */;


-- Dumping database structure for essentialmode
CREATE DATABASE IF NOT EXISTS `essentialmode` /*!40100 DEFAULT CHARACTER SET utf8mb4 COLLATE utf8mb4_bin */;
USE `essentialmode`;

-- Dumping structure for table essentialmode.users
CREATE TABLE IF NOT EXISTS `users` (
  `identifier` varchar(50) COLLATE utf8mb4_bin DEFAULT NULL,
  `license` varchar(50) COLLATE utf8mb4_bin DEFAULT NULL,
  `money` int(11) DEFAULT NULL,
  `bank` int(11) DEFAULT NULL,
  `permission_level` int(11) DEFAULT NULL,
  `group` varchar(50) COLLATE utf8mb4_bin DEFAULT NULL
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin;

-- Data exporting was unselected.
/*!40101 SET SQL_MODE=IFNULL(@OLD_SQL_MODE, '') */;
/*!40014 SET FOREIGN_KEY_CHECKS=IF(@OLD_FOREIGN_KEY_CHECKS IS NULL, 1, @OLD_FOREIGN_KEY_CHECKS) */;
/*!40101 SET CHARACTER_SET_CLIENT=@OLD_CHARACTER_SET_CLIENT */;

**Comment by Rikydc01** [comment]
Of course now it works better than before... So it was just to change the name of database because mine wasn't equals to essentialmode so now it works 👍

**Comment by OsakaMines** [comment]
From the fxmigrant readme....
Currently (2019-05-25), it's unlikely for Linux servers to work with this as-is, this might need either a new artifact build or the usual copying of CitizenFX.Core.Server.dll.


Also, the fxmigrant github is source only, no release builds.  Linux users will have to remove references to fxmigrant from __resource.lua and manually create the database/table using script above from BigDaddyGrimm...

**Comment by Rikydc01** [comment]
Sorry, so what i have to do in my linux server? Because it doesn't work!
![image](https://user-images.githubusercontent.com/49121547/58637331-95e4b100-82f2-11e9-8b81-0dd10b9d06cd.png)

**Comment by paulvilla** [comment]
I have this error.
I donwload fxmigrant but this crash server

Images "out fxmigrant":
![2](https://user-images.githubusercontent.com/14887229/58700322-fc73d880-839f-11e9-93de-b1dabba31b44.png)


Images "with fxmigrant":
![error](https://user-images.githubusercontent.com/14887229/58700383-24fbd280-83a0-11e9-80ea-ab8b85641f42.png)

**Comment by DrAceMisanthrope** [comment]
Yeah, seems that this update hasn't provided the new information it needs, so this issue thread is where people need to come to get the info. I have a new Linux server running FiveM and I had the same issues, where I wasted a lot of time trying to find the aforementioned esplugin_mysql.sql that no longer exists, despite the README not being updated.

Then I manually installed fxmigrant, only to find that it doesn't work with anymore for Linux.
I will try the method above, with the custom create table scripts and report back... A little annoyed that a great resource is made difficult through poor editing of README, when the process has changed.

**Comment by FlyingAce015** [comment]
I can confirm that if you do as instructed by the helpful users in this thread, you should get it working!

1. Change `dependency 'fxmigrant'` to `--dependency 'fxmigrant'` in resources/esplugin_mysql/_resource.lua (Comment out or just remove it)
2. Enter the following commands when logged into mysql;
    - `CREATE DATABASE IF NOT EXISTS `essentialmode`;`
    - `USE `essentialmode`;
    - `CREATE TABLE IF NOT EXISTS `users` (`identifier` varchar(50) COLLATE utf8mb4_bin DEFAULT NULL, `license` varchar(50) COLLATE utf8mb4_bin DEFAULT NULL, `money` int(11) DEFAULT NULL, `bank` int(11) DEFAULT NULL, `permission_level` int(11) DEFAULT NULL, `group` varchar(50) COLLATE utf8mb4_bin DEFAULT NULL) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin;`

Then run your server after adding `start esplugin_mysql` and `set es_enableCustomData 1`.
This avoids using the broken fxmigrant all together.

**Comment by DrAceMisanthrope** [comment]
I can confirm that if you do as instructed by the helpful users in this thread, you should get it working!

1. Comment out / remove `dependency 'fxmigrant'` in `resources/esplugin_mysql/_resource.lua`
2. Enter the following commands when logged into mysql;
    - CREATE DATABASE IF NOT EXISTS \`essentialmode\`;
    - USE \`essentialmode\`;
    - CREATE TABLE IF NOT EXISTS \`users\` (\`identifier\` varchar(50) COLLATE utf8mb4_bin DEFAULT NULL, \`license\` varchar(50) COLLATE utf8mb4_bin DEFAULT NULL, \`money\` int(11) DEFAULT NULL, \`bank\` int(11) DEFAULT NULL, \`permission_level\` int(11) DEFAULT NULL, \`group\` varchar(50) COLLATE utf8mb4_bin DEFAULT NULL) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin;

**NOTE:** Don't forget to ensure those special quotes around database, table and field names!

Then run your server after adding `start esplugin_mysql` and `set es_enableCustomData 1`.
This avoids using the broken fxmigrant all together.

**Comment by syontics** [comment]
im a little confused with all of this could someone walk me through it?

**Comment by Psycholiquid71** [comment]
Thank you Flying aces was trying to figure out this last night and kept importing an old DB to get it working but want to start from scratch.

**Comment by 99kr** [comment]
> For this new version you only need the resource: fx migrant. No sql files are needed
> […](#)
> Op 26 mei 2019 om 14:30 heeft JasonLee120606 ***@***.***> het volgende geschreven: my esplugin_mysql.sql is missing any one can help? — You are receiving this because you are subscribed to this thread. Reply to this email directly, view it on GitHub, or mute the thread.

Update the installation tutorial then?

**Comment by WillOneTwo2019** [comment]
> 
> 
> I can confirm that if you do as instructed by the helpful users in this thread, you should get it working!
> 
>     1. Comment out / remove `dependency 'fxmigrant'` in `resources/esplugin_mysql/_resource.lua`
> 
>     2. Enter the following commands when logged into mysql;
>        
>        * CREATE DATABASE IF NOT EXISTS `essentialmode`;
>        * USE `essentialmode`;
>        * CREATE TABLE IF NOT EXISTS `users` (`identifier` varchar(50) COLLATE utf8mb4_bin DEFAULT NULL, `license` varchar(50) COLLATE utf8mb4_bin DEFAULT NULL, `money` int(11) DEFAULT NULL, `bank` int(11) DEFAULT NULL, `permission_level` int(11) DEFAULT NULL, `group` varchar(50) COLLATE utf8mb4_bin DEFAULT NULL) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin;
> 
> 
> **NOTE:** Don't forget to ensure those special quotes around database, table and field names!
> 
> Then run your server after adding `start esplugin_mysql` and `set es_enableCustomData 1`.
> This avoids using the broken fxmigrant all together.

There is also a "roles" column that needs to be added to the users table.

**Comment by kanersps** [comment]
FXMigrant should work fine, if it doesn’t the above mentioned sql file should do the trick

**Comment by TheRealDadbeard** [comment]
yeah if you start with a new database nothing will be created, and no documentation that I need another plugin, why?

**Comment by WillOneTwo2019** [comment]
Yes, but my point was that the sql is missing "roles" column which is
included in the migration file. Not sure if it matters though.

**Comment by lfuelling** [comment]
fyi, the last available version of the sql file is [here](https://github.com/kanersps/esplugin_mysql/blob/e2ff44e0514dbfd6752cf345c8a011132851cc2a/esplugin_mysql.sql).

To introduce the changes mentioned by @WillOneTwo2019, it's sufficient to change the `CREATE TABLE IF NOT EXISTS 'users'` part of the script to the following:

```
CREATE TABLE IF NOT EXISTS `users`
(
    `identifier`       varchar(50) COLLATE utf8mb4_bin NOT NULL,
    `license`          varchar(50) COLLATE utf8mb4_bin NOT NULL DEFAULT '',
    `money`            int(11)                         NOT NULL DEFAULT 0,
    `bank`             int(11)                         NOT NULL DEFAULT -1,
    `permission_level` int(11)                         NOT NULL DEFAULT 0,
    `group`            varchar(500) COLLATE utf8mb4_bin NOT NULL DEFAULT '',
    `roles`            varchar(500) COLLATE utf8mb4_bin DEFAULT NULL,
    PRIMARY KEY (`identifier`)
) ENGINE = InnoDB
  DEFAULT CHARSET = utf8mb4
  COLLATE = utf8mb4_bin;
```

**Comment by w33ts** [comment]
Some of us run our servers on Linux, so FXMigrant isn't an option. I'll fork this for a non-FXMigrant version.

**Comment by lfuelling** [comment]
In the case that this repo doesn't support Linux, it should be made clear in the readme. 

Otherwise it would be nice to have *one* project that works across both systems.

**Comment by x0Z3ro0x** [comment]
Everything works as intended, but you do need the sql import. Without it you will throw nothing but errors at startup. Thankfully we have some pretty awesome users here and all the info anyone needs to get it working has already been posted :)

**Comment by kanersps** [comment]
This resource works fine in Linux, however FXMigrant seems to have issues, something that i can’t fix


________________________________

**Comment by DrAceMisanthrope** [comment]
> This resource works fine in Linux, however FXMigrant seems to have issues, something that i can’t fix

But you can update your README.md so that people stop getting confused trying to install something that no longer exists. It tells you to run the SQL, not to use FXMigrant. Please update instructions when they are changed through code restructuring. Thank you!

**Comment by ghost** [comment]
https://github.com/kanersps/esplugin_mysql/pull/10

**Comment by kanersps** [comment]
Added back for Linux users
